pysph/cpy/translator.py

```diff
@@ -693,3 +694,141 @@ def __init__(self, detect_type=ocl_detect_type, known_types=None):
 
     def _get_self_type(self):
         return KnownType('GLOBAL_MEM %s*' % self._class_name)
+
+
+class CUDAConverter(CConverter):
```

Comment:
Please try to reuse as much code as possible, derive this from `OpenCLConverter` so we don't have to update things in two places.